### ISSUE TITLE
Can't connect to octoprint over SSL

### ISSUE THREAD

**Issue by taxilian** [alexrj/Slic3r#3203]
I have my octoprint server exposed via an SSL port so I can access it remotely, but slic3r can't use it.  Is this something that can be trivially fixed? Would be cool =]

Awesome tool, keep up the good work.

**Comment by lordofhyphens** [comment]
The connection code, IIRC, uses the Perl cURL library.  Should be fixable
in code w/o much trouble.

"Nothing unreal exists." - Kiri-kin-tha's First Law of Metaphysics.

**Comment by taxilian** [comment]
If you can give me an idea of where to start looking I may be able to fix it myself?

**Comment by lordofhyphens** [comment]
It'll have to wait until I can get in front of the codebase.

"Nothing unreal exists." - Kiri-kin-tha's First Law of Metaphysics.

**Comment by taxilian** [comment]
Meh, I'm currently sitting in a datacenter re-installing servers; as much as I'd like to, I probably shouldn't be playing with my 3d printer right now anyway =]  No rush, just wanted to let you know I'm willing to help as I can to solve the problem.

Richard

**Comment by lordofhyphens** [comment]
Moving to 1.3.1, I'd like to do this in xsgui instead of Perl.

**Comment by keithellis74** [comment]
Probably related to this.  I use OctoPi as installed with the [dietpi](http://dietpi.com/) distro,  By default this opens up the OctoPi instance on port 5000.  Slic3r does not have an option to change the port.  even if the IP address 192.168.xxx.xxx:5000 is used, it appears just to be checking port 80.  The following error message is received.
﻿
![slic3r error](https://user-images.githubusercontent.com/6196195/34633373-bac084ec-f273-11e7-8781-173649afff94.png)
﻿
Can a port option be added please.

**Comment by Stiixxy** [comment]
Just to let everybody know, i started working on this yesterday and got proof-of-concept fix done.
I'm currently in the proces of improving it and making it configurable.  Since @lordofhyphens posted this 



> 
> 
> Moving to 1.3.1, I'd like to do this in xsgui instead of Perl.

Should I still make a pull request with the fix?

**Comment by lordofhyphens** [comment]
@Stiixxy did you have to add a new Perl module for a dependency?

**Comment by Stiixxy** [comment]
@lordofhyphens yes i beleive it required the LWP::Protocol::https module. I'm gonna recheck it when i'm working on it again. I got a https fix and http credentials for basic authentication because that's how my octoprint is setup.
